src/dex/balancer-v2/balancer-v2.ts

```diff
@@ -109,6 +113,8 @@ export class BalancerV2EventPool extends StatefulEventSubscriber<PoolStateMap> {
     'Weighted',
     'LiquidityBootstrapping',
     'Investment',
+    'StablePhantom',
+    'AaveLinear',
```

Comment:
Both of the pools depend on external contract for rate similar to metaStable pools. Hence just tracking balances in events should not be enough. You can disable them from event system by removing them from the list. Hence the state will always be fetched on chain. This can be further improved by tracking the events of the rate contracts so that a complete event system can be built.